controller.py

```diff
@@ -449,6 +450,16 @@ def run_encoding(
             request_util.post_embedding_to_neural_search(
                 request.project_id, embedding_id
             )
+
+        if get_config_value("is_managed"):
+            pickle_path = os.path.join(
+                "/inference", request.project_id, f"embedder-{embedding_id}.pkl"
+            )
+            if not os.path.exists(pickle_path):
+                os.makedirs(os.path.dirname(pickle_path), exist_ok=True)
+                with open(pickle_path, "wb") as f:
+                    pickle.dump(embedder, f)
```

Comment:
would it be possible to have an endpoint to store this? Not sure what happens with the embedded but recreating the whole embedding just to store it feels like a lot of time for 50k projects.

Reply:
The critical part of storing the embedder is the included data for the Principal Component Analysis, which is used to reduce the dimensionality of the created embeddings. The PCA is calculated on a subset of the embeddings (I think the first 5 batches with 128 embeddings each). But I don't think we would know which embeddings have been used for the PCA. However, I believe the deviation by choosing the embeddings for the PCA random would be neglectable.

We would only need to save the PCA for each project (embedding per project). The actual embedders could be shared between projects. This would also save disk space. It is also part of the optimization list.

So, in short, yes this should be possible, but would require some changes.

Reply:
works for me, :)